tlsproxy.go

```diff
@@ -158,3 +158,55 @@ func doRun(idleTimeout time.Duration, l net.Listener, dial func() (net.Conn, err
 		}()
 	}
 }
+
+func dial(keepAlivePeriod time.Duration, forwardAddr string) (net.Conn, error) {
+	conn, err := net.DialTimeout("tcp", forwardAddr, 30*time.Second)
+	if err == nil {
+		addKeepalive(keepAlivePeriod, conn)
+	}
+	return conn, err
+}
+
+func addKeepalive(keepAlivePeriod time.Duration, conn net.Conn) net.Conn {
+	if keepAlivePeriod <= 0 {
+		return conn
+	}
+
+	c, ok := conn.(*net.TCPConn)
+	if !ok {
+		log.Error("Conn was not a TCPConn, can't set KeepAlivePeriod!")
+		return conn
+	}
+	c.SetKeepAlivePeriod(keepAlivePeriod)
```

Comment:
http://felixge.de/2014/08/26/tcp-keepalive-with-golang.html says that keepalive is disabled by default. I guess `SetKeepAlive(true)` should be called too.